Fabric.Authorization.AccessControl/src/app/services/current-user.service.ts

```diff
@@ -17,10 +17,13 @@ export class CurrentUserService {
   getPermissions(securableItem: string): Observable<string[]> {
     const currentTimeStamp = new Date().getTime();
     if (this.shouldRefreshCache(securableItem, currentTimeStamp)) {
+      return this.authUserService.getCurrentUserPermissions(securableItem).pipe(
+         tap(userPermissionResponse => {
           this.lastUpdateTimestampMap.set(securableItem, currentTimeStamp);
```

Comment:
Moved this section into the observable, so the cache should hopefully not be returned until a value is returned.